streaming/aggregate/max.go

```diff
@@ -0,0 +1,95 @@
+package aggregate
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/cube2222/octosql"
+	"github.com/cube2222/octosql/streaming/storage"
+)
+
+var currentMaxPrefix = []byte("$current_max$")
+
+type Max struct {
+}
+
+func NewMaxAggregate() *Max {
+	return &Max{}
+}
+
+func (agg *Max) AddValue(ctx context.Context, tx storage.StateTransaction, value octosql.Value) error {
```

Comment:
Wouldn't we want to only allow Max to store values of the same type? As in we wouldn't want someone to get Max from a float, int and a string, since it makes no sense. (I mean it's their fault that they have that sort of data, but maybe you could add some type checking here)

Reply:
I can store the type of first incoming value and then check the type of it every time AddValue or RetractValue is called, but it might be one topic to discuss (@cube2222 )

Reply:
Both approaches are valid really. If sorting between types sorts first by type (which it fits) then it kind of makes sense. But an error helps catch bugs, and I can't really think of a use case here.

Reply:
sure, will add type checking then

Reply:
The Master told us we don't have to bother :) @cube2222 @JasiekChomiak